src/neps/api.py

```diff
@@ -200,24 +202,30 @@ def run(
         )
         max_cost_total = searcher_kwargs["budget"]
         del searcher_kwargs["budget"]
-    
+
     logger = logging.getLogger("neps")
     logger.info(f"Starting neps.run using root directory {root_directory}")
-    
+
+    # Used to create the yaml holding information about the searcher.
+    # Also important for testing and debugging the api.
+    searcher_info = {
+        "searcher_name": "",
+        "searcher_alg": "",
+        "searcher_selection_source": "",
+        "searcher_modified_arguments": {},
+    }
+
     if isinstance(searcher, BaseOptimizer):
         searcher_instance = searcher
-        searcher_name = "custom"
-        searcher_alg = searcher.whoami()
-        user_defined_searcher = True
+        searcher_info["searcher_name"] = "custom"
```

Comment:
Won't this trigger for every kind of `BaseOptimizer`, even one's where we know their names?

Reply:
True. Searcher name is currently the yaml file name where the searcher arguments and the algorithm type are located (If a user create a new yaml named `bo_new.yaml`, `searcher_name` would then be `bo_new` and `searcher_algorithm` would be `BayesianOptimization`). For BaseOptimizer case, since the searcher is already passed as an instance, the adjustment to searcher_name being `custom` was done.